subsys/settings/src/settings_nvs.c

```diff
@@ -178,7 +237,20 @@ static int settings_nvs_save(struct settings_store *cs, const char *name,
 			continue;
 		}
 
-		if ((delete) && (name_id == cf->last_name_id)) {
+		if (!delete) {
+#if CONFIG_SETTINGS_NVS_CACHE
+			settings_nvs_cache_add(cf, name, name_id);
+#endif
+			write_name_id = name_id;
+			write_name = false;
+		}
+
+		goto found;
+	}
+
+found:
+	if (delete) {
```

Comment:
Note that this code is now executed not only if `goto found` is called, but also if the while loop breaks, e.g. because of `if (name_id == NVS_NAMECNT_ID)` condition. Won't it cause removal of `last_name_id` by the line `rc = nvs_delete(&cf->cf_nvs, name_id);` below?

Reply:
Can you give me an example? I really don't think there are any problems. I think the current changes are equivalent to the past logic, which I want to make every effort to ensure.

Reply:
An example is that if you try to delete a non-existing entry the while loop breaks and you end up with `nvs_delete(&cf->cf_nvs, name_id)` where `name_id == NVS_NAMECNT_ID`. That ID is used to store `last_name_id`, so either something will not work correctly on the next boot or NVS layer will need to regenerate `last_name_id`. Either way, the logic doesn't seem equivalent.

Reply:
Yes, I see. Thank you.